### PR TITLE
Clipboard: Adapt to new `make` pattern + improve tests

### PR DESCRIPTION
**Pull request by davedbase**
- [x] Finalize API design
- [x] Make functional minimal demo
- [x] Port and validate tests
- [x] Update README and package.json

### CODE REVIEW

File: packages/clipboard/src/index.ts [hunk 1]
```diff
@@ -26,11 +26,11 @@ declare module "solid-js" {
...
  * ```ts
- * const [ write, read ] = createClipboard();
+ * const [ write, read ] = makeClipboard();
  * console.log(await read());
  * ```
  */
-const createClipboard = (): [
+export const makeClipboard = (): [
   write: ClipboardSetter,
   read: () => Promise<string>,
```
Comment (thetarnav) [inline packages/clipboard/src/index.ts:35]: Should we take this opportunity to switch [write, read, something_else] to [read, write, something_else]?
Maybe writing is more common operation, but [read, write] is a more common api.

**Review by thetarnav — COMMENTED** [review]
`makeClipboard` seams weird... It doesn't actually do anything, functions that it provides could be exported separately.
But then again... Those functions separately, wouldn't do anything special either. Just alias the browser API. Maybe this pattern just doesn't fit here? Don't know if we should export functions that don't have a clear purpose.

File: packages/clipboard/src/index.ts [hunk 2]
```diff
@@ -44,6 +44,31 @@ const createClipboard = (): [
   return [write, read, { newItem }];
 };
 
+/**
+ * Creates a new reactive primitive for managing the clipboard.
+ *
+ * @return write - A method to write to the clipboard
+ * @return read - Read from the clipboard
+ * @return helpers - helper functions:
```
Comment (thetarnav) [inline packages/clipboard/src/index.ts:52]: old

Comment (davedbase) [reply packages/clipboard/src/index.ts:52]: Yeah I haven't adjusted ANY of these yet. Just focusing on the API first.

File: packages/clipboard/src/index.ts (hunk 1 above)
Comment (davedbase) [reply packages/clipboard/src/index.ts:35]: Yup I'll be sure to do that, good point.

**Comment by davedbase** [comment]
@thetarnav I had another look at makeClipboard and I think even though it doesn't do any clean-up I think it's important to wrap this base functionality. It's used in the directive and the reactive primitive. I like that we're wrapping ClipboardItem as well with newItem.

**Comment by davedbase** [comment]
I think I've taken this as far as I can at this point. I'd like to get it up and improve/finish the tests at another date. Would love a look and approval to merge @thetarnav @atk :-)

File: packages/clipboard/README.md [hunk 3]
```diff
@@ -21,27 +21,119 @@ yarn add @solid-primitives/clipboard
...
+];
+```
+
 ### createClipboard
 
-Clipboard exports a read and write function. Note the write function is exported first for convenience as the most common use case for this primitive.
+This primitive provides full facilities for reading and writing to the clipboard. It allows for writing to clipboard via exported function or input signal. It wraps the Clipboard Async API with a resource and supplies reactive helpers to make pulling from the clipboard easy.
+
+```tsx
+const [data, setData] = createSignal('Hello);
```
Comment (thetarnav) [inline packages/clipboard/README.md:47]: Lol must have missed the notification. ehh
Found a typo though. The missing `'` is messing the code highlighting.

**Pull request merged**